File: packages/common/erp-common-ms-0-js/src/servers/graphql/ms/3/country.ts
import { ICountry, ICreateCountry, IUpdateCountry, IDeleteCountry, ISearchCountry } from '@gmahechas/erp-common';
import { axiosClient, env } from '@gmahechas/erp-common-ms-utils-js';

export const createOneCountry = async (data: ICreateCountry): Promise<ICountry> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: ICountry }>('/rest/v1/3/country/create/one', data);
	return response.data;
};

export const updateOneCountry = async (data: IUpdateCountry): Promise<ICountry | null> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).put<{ data: ICountry }>('/rest/v1/3/country/update/one', data);
	return response.data;
};

export const deleteOneCountry = async (data: IDeleteCountry): Promise<ICountry | null> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).delete<{ data: ICountry }>('/rest/v1/3/country/delete/one', { data });
	return response.data;
};

export const searchOneCountry = async (data: Partial<ISearchCountry>): Promise<ICountry | null> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: ICountry }>('/rest/v1/3/country/search/one', data);
	return response.data;
};

export const searchManyCountry = async (data: Partial<ISearchCountry>[]): Promise<ICountry[]> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: ICountry[] }>('/rest/v1/3/country/search/many', data);
	return response.data;
};

File: packages/3/ms-3-gingo/routes/v1/country.go
package routes

import (
	ginGoMiddlewares "github.com/gmahechas/erp-common-gin-go/middlewares"
	ginGoUtils "github.com/gmahechas/erp-common-gin-go/utils"
)

func InitCountryRoutes(router *ginGoUtils.GinRouterGroup) {
	routes := router.Group("/country")
	{
		routes.GET("/create/one", ginGoMiddlewares.AuthenticationMiddleware(), ginGoMiddlewares.AuthorizationMiddleware(), ginGoMiddlewares.ValidatorMiddleware(), ginGoUtils.GinHandlerFunc(createOneCountry))
		routes.GET("/update/one", ginGoUtils.GinHandlerFunc(updateOneCountry))
		routes.GET("/delete/one", ginGoUtils.GinHandlerFunc(deleteOneCountry))
		routes.GET("/search/one", ginGoUtils.GinHandlerFunc(searchOneCountry))
		routes.GET("/search/many", ginGoUtils.GinHandlerFunc(searchManyCountry))
	}
}

func createOneCountry(context *ginGoUtils.GinContext) {
	context.String(200, "createOneCountry !!!")
}

func updateOneCountry(context *ginGoUtils.GinContext) {
	context.String(200, "updateOneCountry !!!")
}

func deleteOneCountry(context *ginGoUtils.GinContext) {
	context.String(200, "deleteOneCountry !!!")
}

func searchOneCountry(context *ginGoUtils.GinContext) {
	context.String(200, "searchOneCountry !!!")
}

func searchManyCountry(context *ginGoUtils.GinContext) {
	context.String(200, "searchManyCountry !!!")
}
